rtmp-server/src/server.rs

```diff
@@ -0,0 +1,75 @@
+use crate::{error::RtmpError, handshake::Handshake};
+use std::{net::SocketAddr, time::Duration};
+use tokio::{
+    io::{AsyncRead, AsyncWrite},
+    net::TcpListener,
+    time::timeout,
+};
+use tracing::{error, info};
+
+#[allow(dead_code)] // TODO add SSL/TLS
+pub struct ServerConfig {
+    pub port: u16,
+    pub use_ssl: bool,
+    pub certfile: Option<String>,
+    pub keyfile: Option<String>,
+    pub cacertfile: Option<String>,
+    pub client_timeout_secs: u64,
+}
+
+pub struct RtmpServer {
+    config: ServerConfig,
+}
+
+impl RtmpServer {
+    pub fn new(config: ServerConfig) -> Self {
+        Self { config }
+    }
+
+    pub async fn run(&self) -> Result<(), RtmpError> {
+        let addr = SocketAddr::from(([0, 0, 0, 0], self.config.port));
+        let listener = TcpListener::bind(addr).await?;
+
+        info!("RTMP server listening on port {}", self.config.port);
+
+        loop {
+            match listener.accept().await {
+                Ok((stream, peer_addr)) => {
+                    info!("New connection from: {}", peer_addr);
+
+                    let client_timeout = self.config.client_timeout_secs;
+
+                    tokio::spawn(async move {
+                        // use stream or tls_stream based on use_ssl field
+                        let result = handle_client(stream, client_timeout).await;
+
+                        if let Err(e) = result {
+                            error!("Client handler error: {}", e);
```

Comment:
I think it would be better to do
```suggestion
                            error!(error = %e, "Client handler error");
```